docs/language/functions.mdx

```diff
@@ -516,3 +479,28 @@ fun incrementN() {
 }
 ```
 
+## Functions are Values
+
+Functions are values ("first class"), so e.g. may be assigned to variables and fields,
+or passed to functions as arguments.
+
+
+```cadence
+// Declare a function named `transform` which applies a function to each element
```

Comment:
In most other languages this would be called `map`. IMO it is unnecessarily confusing to use a different name for this here. Since this only functions on integers, perhaps `mapIntegers`?

Reply:
Haha, I intentionally didn't use the "functional programming" term to not confuse developers with "maps", as in the alternative name for dictionaries in some languages.

Maybe there should be a more concrete example here.